### DISCUSSION TITLE
enable dragging of only 1 shape

### DISCUSSION THREAD

**Discussion by cubecleveland** [geoman-io/leaflet-geoman#1083] [Questions & Help] [answered]
Hi,

First, thanks for the awesome answers and support.
I browsed through the docs and it seems like not posible but is there a way to enable dragging mode per layer ?

**Comment by Falke-Design** [comment, marked as answer]
https://github.com/geoman-io/leaflet-geoman#drag-mode:
```
layer.pm.enableLayerDrag();
```

**Reply by Falke-Design** [reply]
Can I ask you on what project you are working on? Maybe you want to share it here #904

**Comment by cubecleveland** [comment]
Absolutly. We are wokring ona  complete editable map with location drops and physical device location distributon. at the moment we are working on the logic part. sicne we are using React its a little hard to get the logic right and the state work and we are having some issue with the state management coming from Geoman. 
We are leaning toward keeping the Edited map stateless and inject the finsihed state ti a react comp -- bu that yet to be solved.

We might need your help again :)

I will keep you posted as we go along.

We already made up our mind to use the plugin so we will evetually try (unless problem come that can not be solved) to work around the issues.
Again, your help is amaizng.
thanks!

**Comment by cubecleveland** [comment]
I have noticed in your wonderfule example in isseu #904 u have an example of adding an arrow, is that a special plugin you are using ? or can that be done vie geoman ?

**Reply by Falke-Design** [reply]
I have created it by my self based on Geoman. But it has some bugs while dragging / editing.

Here the code if you want:
[Arrow-Geoman.js.txt](https://github.com/geoman-io/leaflet-geoman/files/7912623/Arrow-Geoman.js.txt)

**Comment by cubecleveland** [comment]
Many thanks, can i ask you what platform you used to build this app ? (React, Angular) etc.. it will help me undrstand how o manage the state ....

**Reply by Falke-Design** [reply]
Vanilla JavaScript and then building with wepack

**Reply by cubecleveland** [reply]
Follow up, when adding markers i want to add special attributes to the individual markers.

Currently i use My own method wheer i call enableDraw with Marker..... i can only pass markerStyle... how will i be able to set a special property on the specific marker ...... 

The use case is that those marker are going to a dtabase and i need an id or token in the property to put them back in the map with their relative info

**Reply by Falke-Design** [reply]
You get the new added markers in the `map.on('pm:create',(e)=>{})` event where you can add properties to them

**Reply by cubecleveland** [reply]
Do u mind sharing the plugin in u used or how u handled the pattern of the polygon in your app ?

**Comment by cubecleveland** [comment]
I found a better solution:
```
handleDrawShape("Marker", {
            markerStyle: { icon, properties: { someId : id } },
            continueDrawing: false,
          });
```
to whoever is interested

**Comment by cubecleveland** [comment]
is there any way to set the tootip message to a custom one when for example placing a circle Marker on the map. To a message like 'Place Item here' ?
